shell/key-bindings.zsh

```diff
@@ -128,25 +128,51 @@ fi
 
 # CTRL-R - Paste the selected command from history into the command line
 fzf-history-widget() {
-  local selected
-  setopt localoptions noglobsubst noposixbuiltins pipefail no_aliases noglob nobash_rematch 2> /dev/null
+  local selected extracted_with_perl=0
+  setopt localoptions noglobsubst noposixbuiltins pipefail no_aliases no_glob no_ksharrays extendedglob 2> /dev/null
```

Comment:
Option changes remarks:
- `no_ksharrays` is needed. By default, array elements are numbered from one in
  zsh; when a user sets `ksharrays`, array elements are numbered from zero
  - I believe 3a6af27586c6 was the right approach, and reverting it was a
    mistake, but the original author deleted their commit post so I can't see what the
    actual issue was. `emulate -L zsh` would therefore automatically set the
    default array behavior, and there would be no need for setting `no_ksharrays`
- `no_glob` and `extendedglob` do not conflict with one another and serve
  different purposes (man zshoptions)
- `extendedglob` provides additional pattern‑matching syntax (such as `#` or
  flags like `(#b)`)
- `nobash_rematch` was only useful for `=~` usage